File: ios-client/KeyPair.m
//
//  KeyPair.m
//  Keysmith IOS Client
//
//  Created by Hakan D on 04/05/2013.
//  Copyright (c) 2013 Hakan Dilek. All rights reserved.

#import "KeyPair.h"

@implementation KeyPair

@synthesize publicKey, privateKey;

- (id) init: (PublicKey*) pubKey with: (PrivateKey*) priKey;{
    self = [super init];
    if (self) {
        publicKey = pubKey;
        privateKey = priKey;
        [publicKey retain];
        [privateKey retain];
    }
    return self;
}

- (void) dealloc {
    [publicKey release];
    [privateKey release];
    [super dealloc];
}

@end

File: dotnet-client/Keysmith.Client.Lib/KeyPair.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="KeyPair.cs" company="Hakan Dilek">
//   (c) 2013 Hakan Dilek
// </copyright>
// <summary>
//   The key pair.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace Keysmith.Client.Lib
{
    /// <summary>
    /// The key pair.
    /// </summary>
    public class KeyPair
    {
        /// <summary>
        /// Gets or sets the public key.
        /// </summary>
        public PublicKey PublicKey { get; set; }

        /// <summary>
        /// Gets or sets the private key.
        /// </summary>
        public PrivateKey PrivateKey { get; set; }
    }
}